components/clarity-lsp/src/common/state.rs

```diff
@@ -342,9 +344,14 @@ pub async fn build_state(
         }
     };
 
-    let (deployment, mut artifacts) =
-        generate_default_deployment(&manifest, &StacksNetwork::Simnet, false, file_accessor)
-            .await?;
+    let (deployment, mut artifacts) = generate_default_deployment(
+        &manifest,
+        &StacksNetwork::Simnet,
+        false,
+        file_accessor,
+        Some((StacksEpochId::Epoch21, ClarityVersion::Clarity2)),
```

Comment:
I think this should be `Clarity1` until 2.1 is live, so that contracts are checked in that context. It will still use the new parser.

Reply:
So I ended up adding an argument only for the Epoch

Reply:
Sounds good. It should probably be `Epoch2_05`, but I'm not sure if that will matter or not.

This brings a question to mind: how will we handle this later, after 2.1 is live. How do we decide which epoch and clarity version to use for the dependencies?

Reply:
By `It should probably be `Epoch2_05`, I meant on line 381.